artifactory/services/delete.go

```diff
@@ -122,3 +130,30 @@ func (ds *DeleteParams) SetIncludeDirs(includeDirs bool) {
 func NewDeleteParams() DeleteParams {
 	return DeleteParams{ArtifactoryCommonParams: &utils.ArtifactoryCommonParams{}}
 }
+
+func removeDirsContainArtifactsWithPropsExcluded(specFile utils.ArtifactoryCommonParams, ds *DeleteService, deleteCandidates []utils.ResultItem) ([]utils.ResultItem, error) {
```

Comment:
I sugest adding the following description to this function:

*This function receives as an argument the list of files and folders to be deleted from Artifactory.
In case the search params used to create this list included excludeProps, then we might need to remove some directories from this list. These directories are removed from this list because they must not be deleted from Artifactory.
These directories must be deleted, because they include files, which should not be deleted, because of the excludeProps params.*